stock_picking_product_generate_sequence/models/product_template.py

```diff
@@ -0,0 +1,56 @@
+##############################################################################
+# Copyright (c) 2022 brain-tec AG (https://braintec.com)
+# All Right Reserved
+#
+# See LICENSE file for full licensing details.
+##############################################################################
+from re import match as regex_match, split as regex_split
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    serial_lot = fields.Char(
+        string="Sequence Product",
+        help="Create a sequence for product. Remember sequence contains one character "
+        "block and one numeric block. For the numeric part of the sequence, 0 must be "
+        "used to create the padding. E.g.: TEST/00000 (padding is 5)",
+    )
+
+    @api.onchange("serial_lot")
+    def _onchange_serial_lot(self):
+        pattern = "(?:^\\D{1,}\\d{1,}$)"
+        split_sequence_product = regex_split(r"(\d+)", self.serial_lot)
```

Comment:
In my test, I get an error here if `self.serial_lot` is not set (`False`).
Is it working on your end? Or is it not possible that `self.serial_lot` is False?

I just tried to create a new product.

Reply:
Good input, I think I lost sight of product creation. 
But really, if there is no sequence in the product, the global sequence is used.
I added a condition that only checks if a sequence exists.